src/app/zap-templates/zcl/data-model/chip/test-cluster.xml

```diff
@@ -520,5 +550,16 @@ limitations under the License.
       <field id="5" name="arg5" type="SimpleStruct" array="true"/>
       <field id="6" name="arg6" type="SimpleEnum" array="true"/>
     </event>
+
+    <event code="0x0002" name="TestFabricScopedEvent" priority="info" side="server">
+      <description>Example test event</description>
+      <field id="1" name="arg1" type="fabric_idx"/>
+    </event>
+
+    <event code="0x0003" name="TestNullableFabricScopedEvent" priority="info" side="server">
+      <description>Example test event</description>
+      <field id="1" name="arg1" type="fabric_idx" isNullable="true"/>
```

Comment:
What spec construct does this correspond to.  Why are we caring about this case?

Reply:
For events in ACL cluster, it has FabricIndex, we generate code FabricIndex GetFabricIndex() const { return adminFabricIndex; } where adminFabricIndex is not nullable type, its type is FabricIndex.
For events in doorlock cluster, it has nullable fabricindex, we generate DataModel::Nullable<chip::FabricIndex>  etFabricIndex() const { return fabricIndex; }  where fabricindex is not nullable type. 
Then for LogEvent, I wrote two kind of template function to correspond to these two situation, so in order to validate these two situation, I have type="fabric_idx" isNullable="true".

This is nullable fabric index from doorlock cluster
https://github.com/project-chip/connectedhomeip/blob/master/src/app/zap-templates/zcl/data-model/chip/door-lock-cluster.xml#L627

Reply:
> For events in doorlock cluster, it has nullable fabricindex

Those are NOT FABRIC SCOPED events.  They are just events that happen to have a fabricIndex that does NOT scope them, and they must be readable by all fabrics.  Which fabric-scoped events are not.

Reply:
Per offline chat, we would check for a field that is of type fabric_idx and not nullable which could be used for fabric-scoped event, currently acl is fabric scope one. updated the code